include/dlaf_c/eigensolver/eigensolver.h

```diff
@@ -0,0 +1,102 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2023, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#include <dlaf_c/desc.h>
+#include <dlaf_c/utils.h>
+
+#ifdef DLAF_WITH_SCALAPACK
+
+/// Eigensolver
+///
+/// @remark This function is available only when DLAF_WITH_SCALAPACK=ON.
+///
+/// @pre The matrices \f$\mathbf{A}\f$ and \f$\mathbf{Z}\f$ are assumed to be distributed and in host
+/// memory. The vector of eigenvalues \f$\mathbf{w}\f$ is assumed to be local (non-distributed) and in
+/// host memory. Moving to and from GPU memory is handled internally.
+///
+/// @pre Submatrices are currently not supported, so @p n is the size of the full matrices
+/// \f$\mathbf{A}\f$ and \f$\mathbf{Z}\f$ and @p ia, @p ja, @p iz, and @p jz need to be 1.
+///
+/// @post The pika runtime is resumed when this function is called and suspended when the call
+/// terminates.
+///
+/// @param uplo indicates wheather the upper ('U') or lower ('L') triangular part of the global submatrix
+/// \f$\mathbf{A}\f$ is referenced
+/// @param n order of the sumbatrix \f$\mathbf{A}\f$ used in the computation
+/// @param a Local part of the global matrix \f$\mathbf{A}\f$
+/// @param ia row index of the global matrix \f$\mathbf{A}\f$ identifying the first row of the submatrix
+/// $A$, has to be 1
+/// @param ja column index of the global matrix \f$\mathbf{A}\f$ identifying the firs column of the
+/// submatrix \f$\mathbf{A}\f$, has to be 1
+/// @param desca ScaLAPACK array descriptor of the global matrix \f$\mathbf{A}\f$
+/// @param w Local vector of eigenvalues (non-distributed)
+/// @param z Local part of the global matrix \f$\mathbf{Z}\f$
+/// @param iz row index of the global matrix \f$\mathbf{Z}\f$ identifying the first row of the submatrix
+/// \f$\mathbf{Z}\f$, has to be 1
+/// @param jz column index of the global matrix \f$\mathbf{A}\f$ identifying the firs column of the
+/// submatrix \f$\mathbf{A}\f$, has to be 1
+/// @param descz ScaLAPACK array descriptor of the global matrix \f$\mathbf{Z}\f$
+/// @param info 0 if the eigensolver completed normally
```

Comment:
probably is clear for who is used to this, but... might it be worth specifying it is a return value?

Reply:
`@return` does not take a name, I think the best option is `@param[out]`?

Reply:
I'm not super proficient with oxygen, but it looks like the right choice 😉